protocol/0088-TRTO-trading_transaction_ordering.md

```diff
@@ -0,0 +1,41 @@
+
+## Execution Ordering
+
+Trading transactions (those which create, update or remove orders of any type on any market) should be executed in a specific way once included in a block. This ordering is per-market (inter-market ordering is unspecified) and should be enableable at a market configuration level (i.e. it can be enabled for some markets and not for others). For a market where this is not enabled, all transactions on the market enact in the order specified in the block itself.
+
+Chiefly, when enabled all transactions which would cancel an order or create post-only orders should be executed first before transactions which could create trades. The ordering of the transactions in this way means that, at the time of each block being created, all parties who are contributing to creating an order book have an opportunity to update their prices prior to anyone who would capitalise on temporary stale prices, regardless of which transaction reached the validator's pre-block transaction pool first. This ordering can still be seen to be a "fair" transaction ordering, as parties cannot take actions which would cause a trade, but only take action to avoid trading at a price they no longer desire (or indeed to improve a price prior to trade-creating transactions' arrival).
+
+Furthermore, transactions which can cause a trade by acting aggressively, such as non-post-only orders and amends, will be delayed by one block prior to execution. This results in the pattern where:
+
+ 1. Prior to block N, post only order A and market order B arrive to the chain, these are both included in block N.
+ 1. When block N is executed, order A takes effect.
+ 1. Prior to block N + 1, post only order C then market order D and finally a cancellation of order A arrive to the chain, these are both included in block N + 1.
+ 1. When block N + 1 is executed, order C first takes effect, then the cancellation of order A, then finally market order B.
+ 1. When block N + 2 is executed, market order D takes effect
+
+## Batch Transactions
+
+Batch transactions, as they contain different order types, must be split apart for the execution of their different components at the specified times. All cancellations, and any post-only order creations, should be separated from amendments and non-post-only orders. The first group will be executed all together within the first section of the inclusion block, whilst the second group will be executed all together within the second section of the block after the inclusion block.
```

Comment:
A possible simplification would be to assign the whole batch to the lowest priority set of any order in the batch (where P1 is cancellations, P2 post-only orders, and P3 anything else), and execute the whole thing then. This would post no issues that I can see.